basic_auth/api/application.py

```diff
@@ -41,8 +45,22 @@ def __init__(self, name, resource):
             self._collection_methods_map)
         payload = await self._validate_request(request, allowed_methods)
 
+        date_format = "%Y-%m-%d-%H-%M"
+
+        start_date = request.query.get('start_date')
+        if start_date is not None:
+            start_date = datetime.strptime(start_date, date_format)
```

Comment:
This could raise a value error and it's based on what's in the request query, so we should be more defensive.
Given the same logic is repeated for end_date, I'd abstract it out in a function like
```python
def date_from_query(request, key):
    value = request.query.get(key)
    if not value:
        return None
    try:
        return datetime.strptime(value, date_format)
    except ValueError:
        return None
```